CV App Project and Refs&Memoization: add descriptive links (#27875)
TheOdinProject/curriculum: react/more_react_concepts/refs_and_memoization.md

<div class="lesson-content__panel" markdown="1">

1. The article [When to useMemo and useCallback](https://kentcdodds.com/blog/usememo-and-usecallback) by Kent C. Dodds further introduces more examples of when to use `useMemo` and `useCallback` and when you shouldn't bother using them.

> 1. We've only learned about a basic implementation of the `useRef` hook. For more examples about its usage and why we should be wary of using the hook (more on the links they provided in the guide), check out the interactive guide of the React documentation for [`useRef` hook](https://react.dev/reference/react/useRef) .

Reviewer: ```suggestion
1. We've only learned about a basic implementation of the `useRef` hook. For more examples about its usage and why we should be wary of using the hook (more on the links they provided in the guide), check out the interactive guide of the React documentation for [useRef hook](https://react.dev/reference/react/useRef) .
```
Forgot to add this to the review as well. Good spot on this since I somehow missed adding it to the spreadsheet.
Your intuition is correct, in this case (2 words, 1 with backticks) the single backticked word makes the link harder to spot/looks like 2 separate links. So in this scenario, just not using backticks looks perfectly fine. With more non-backticked words in the link text, this becomes less of a visual issue.

1. The article [useRef instead of querySelector in React](https://meje.dev/blog/useref-not-queryselector) by Caleb Olojo briefly tells some unexpected behaviors when trying to manipulate the DOM directly with DOM manipulation methods and why we should prefer `useRef` over other DOM manipulation methods like `querySelector`. Check it out!
1. As we have learned, the `useRef` hook has other uses other than what we've primarily covered which is DOM Manipulation. Get to know more about its use-cases in this great article by Dan Abramov [Making setInterval Declarative with React Hooks](https://overreacted.io/making-setinterval-declarative-with-react-hooks/).

</div>

### Knowledge check

The following questions are an opportunity to reflect on key topics in this lesson. If you can't answer a question, click on it to review the material, but keep in mind you are not expected to memorize or master this knowledge.

- [Why should you prefer useRef hook over other DOM manipulation methods like querySelector?](https://meje.dev/blog/useref-not-queryselector)
- [What is the difference between useMemo and useCallback?](#usememo-or-usecallback)

> - [How do useMemo and useCallback help optimize the performance of React components?](#optimization-description)

Author: I've also added this reference in the lesson because the old one didn't point to any resource.

> This section contains helpful links to related content. It isn’t required, so consider it supplemental.

Reviewer: ```suggestion
This section contains helpful links to related content. It isn't required, so consider it supplemental.
```
Literally just fixing a strange alternative apostrophe character to appease the linter.

> - The `memo` function is straightforward that's why we didn't cover it too much, but if you want to know more about it, especially if you want to create your own logic for the `memo` go to the documentation for [memo wrapper function](https://react.dev/reference/react/memo).

Reviewer: Aha, brilliant spot regarding the "same link text for same href" thing! :rocket: 
This definitely does not need duplication in the AR section, so could you replace it with the following please?
```suggestion
- - It looks like this lesson doesn't have any additional resources yet. Help us expand this section by contributing to our curriculum.
```
